### PR TITLE
testing/vault: bumping to 0.6.2

### PR DESCRIPTION
**Pull request by gena01**
* Build latest 0.6.2 release
* Fix "mlock" permission in init.d script so vault can start
  as "vault" user.
* Adding cleanup script to stop the vault service on package
  removal
* Ading cleanup script to remove vault user and group on package
  removal

### CODE REVIEW

File: testing/vault/vault.post-deinstall [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+# Delete vault group and user
+delgroup vault 2> /dev/null
+deluser vault 2> /dev/null
```
Comment (jirutka) [inline testing/vault/vault.post-deinstall]: This is not a good idea. You can’t know if the vault user/group has been created by the package or not; user can use it even for something else.

File: testing/vault/vault.pre-deinstall [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+# Stop vault service
+rc-service vault stop &> /dev/null
```
Comment (jirutka) [inline testing/vault/vault.pre-deinstall]: `&>` is not POSIX-sh compatible, it’s a _bashism_. I also don’t think that it’s needed, user should now that (s)he should stop the service before uninstalling…

File: testing/vault/vault.initd [hunk 3]
```diff
@@ -10,9 +10,10 @@ pidfile="/run/${SVCNAME}.pid"
 
 start_pre() {
 	checkpath -f -m 0644 -o ${SVCNAME}:${SVCNAME} "$VAULT_LOG_FILE"
+	setcap cap_ipc_lock=+ep $(readlink -f $(which vault))
```
Comment (jirutka) [inline testing/vault/vault.initd]: This should not be set in runscript, but in APKBUILD.

File: testing/vault/vault.post-deinstall (hunk 1 above)
Comment (gena01) [inline testing/vault/vault.post-deinstall]: I understand, will take it out. (Actually copied from consul package, might need to fix that there as well)

File: testing/vault/vault.pre-deinstall (hunk 2 above)
Comment (gena01) [inline testing/vault/vault.pre-deinstall]: Well if they don't stop it and they have it running then it keeps on running and then if it's killed and added back again "service vault status" shows crashed. I could update this not to be a bashism.

This one was also copied from consul. Maybe another thing to fix there.

File: testing/vault/vault.initd (hunk 3 above)
Comment (gena01) [inline testing/vault/vault.initd]: Yeah wasn't sure about this one, but it did fix the mlock issue. Let me update the commit.

P.S. "setcap" line came from official docs and is needed is vault is not running as root user. See here: https://www.vaultproject.io/docs/config/

**Comment by gena01** [comment]
Updated. Let me know if this looks better.

File: testing/vault/vault.initd (hunk 3 above)
Comment (jirutka) [inline testing/vault/vault.initd]: I understand why it is used and I didn’t mean that you should not use `setcap` at all, just that it should be set in APKBUILD, not post-install script. abuild and apk handles these attributes (if not for you, then use the latest abuild and apk versions).

Comment (gena01) [inline testing/vault/vault.initd]: @jirutka fixed. Sorry didn't know I could to setcap in APKBUILD just like install.

**Comment by jirutka** [comment]
Merged in f625013a34ed3b3129cacf92a271ebaf15dd1d0b, thanks!

**Pull request closed without merging**